app/jobs/migrate_upload_job.rb

```diff
@@ -0,0 +1,55 @@
+require 'uri'
+require 'digest'
+
+class MigrateUploadJob < ApplicationJob
+  queue_as :low
+
+  def perform(icon_id)
+    icon = Icon.find(icon_id)
+    raise IconNotUploaded, "Icon #{icon.id} is remotely hosted" unless icon.s3_key.present?
```

Comment:
possibly for another PR but: do we want to do this for all icons? we've discussed eliminating remote hosting before, and if this does it anyway?